### PR TITLE
Add an UpdateForV9/10 to reroute cluster state

### PR DESCRIPTION
**Pull request by thecoop**
Reminder to update for v9 and v10

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/reroute/ClusterRerouteResponse.java [hunk 1]
```diff
@@ -43,6 +44,7 @@ public class ClusterRerouteResponse extends ActionResponse implements IsAcknowle
     /**
      * To be removed when REST compatibility with {@link org.elasticsearch.Version#V_8_6_0} / {@link RestApiVersion#V_8} no longer needed
      */
+    @UpdateForV9(owner = UpdateForV9.Owner.DISTRIBUTED_COORDINATION)
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/reroute/ClusterRerouteResponse.java:47]: If I am not mistaken, according to the comment above, this should be updated in `UpdateForV10` as `RestApiVersion.V_8` is not going to be removed in V9

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/reroute/ClusterRerouteResponse.java:47]: Oh, yes, quite right

**Review by idegtiarenko — APPROVED** [review]

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
Yes, this should be `UpdateForV10` but we also should introduce an `UpdateForV9` to suppress the state when using the V9 API.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**